scripts/generate_activity_script/src/script.ts

```diff
@@ -346,29 +325,492 @@ async function withdrawRandomNFT(
+  amount: Coin
+) {
+  const txHash = await IBCTransfer(
+    clientSG,
+    addressFrom,
+    addressTo,
+    channel,
+    port,
+    amount,
+    {
+      denom: "untrn",
+      amount: "2000",
+    }
+  );
+  return {
+    mode: NeutronAction.PROCESS_LSM_SHARES_IBC_TO,
+    txHash: txHash,
+  };
+}
+
+/* Function dedicated to choose the random amount of tokens that we want to send in processLSMShares function
+ * MAX_LSM_PROCESS from .env is the maximum amount of tokens that this function can choose to send
+ * If current balance lower than MAX_LSM_PROCESS, then MAX_LSM_PROCESS is current balance
+ */
+async function randomIBCToTransfer(
+  dropCore: DropCoreClient,
+  neutronWallet: Wallet,
+  addressFrom: string,
+  addressTo: string
+): Promise<ActionLog> {
+  const baseDenomBalance: Coin = await neutronWallet.clientCW.getBalance(
+    addressFrom,
+    BASE_DENOM
+  );
+  /* If here is nothing to send on our balance, then just throw an error
+   */
+  if (Number(baseDenomBalance.amount) === 0) {
+    throw `Nothing to transfer via IBC, ${BASE_DENOM} balance is 0`;
+  }
+
+  const exchangeRate: number = Math.floor(
+    Number(await dropCore.queryExchangeRate())
+  );
+  const minExchangeRate: number = exchangeRate + 1;
+  const config = await dropCore.queryConfig();
+
+  /* min is a minimum amount of tokens that this function can randomly choose from interval
+   * It's either a config.lsm_min_bond_amount or minExchangeRate depends on
+   * What is bigger (the biggest here is a chosen minimum)
+   */
+  const min: number =
+    Number(config.lsm_min_bond_amount) < minExchangeRate
+      ? minExchangeRate
+      : Number(config.lsm_min_bond_amount);
+  if (min > Number(baseDenomBalance.amount)) {
+    throw `Nothing to send via IBC, ${BASE_DENOM} balance is lower then min(${min}) (this value either exchange rate or config.lsm_min_bond_amount)`;
+  }
+  if (min > MAX_LSM_PROCESS) {
+    throw `MAX_LSM_PROCESS lower then min(${min}) (this value either exchange rate or config.lsm_min_bond_amount)`;
+  }
+
+  /* max is a maximum amount of tokens that this function can randomly choose from interval
+   * It's either a MAX_LSM_PROCESS or current BASE_DENOM balance depends on
+   * What is less (the smaller here is a chosen maximum)
+   */
+  const max: number =
+    Number(baseDenomBalance.amount) < MAX_LSM_PROCESS
+      ? Number(baseDenomBalance.amount)
+      : MAX_LSM_PROCESS;
+
+  const randomAmount: number = Math.floor(Math.random() * (max - min) + min);
+
+  /* By default in our case it's always "transfer" port
+   */
+  const res = await IBCToTransfer(
+    neutronWallet.clientSG,
+    addressFrom,
+    addressTo,
+    IBC_CHANNEL_TO,
+    "transfer",
+    {
+      denom: BASE_DENOM,
+      amount: String(randomAmount),
+    }
+  );
+  let txDetails: IndexedTx = await neutronWallet.clientCW.getTx(res.txHash);
+  while (txDetails === null) {
+    await sleep(5000);
+    txDetails = await neutronWallet.clientCW.getTx(res.txHash);
+  }
```

Comment:
```suggestion
  let txDetails: IndexedTx | null = null;
  while (txDetails === null) {
    await sleep(5000);
    txDetails = await neutronWallet.clientCW.getTx(res.txHash);
  }
```
also I think you can export it as a helper for awating tx to be included into a block

Reply:
also please consider timeout

Reply:
It seems like we already have a waitForTx, added it